src/packet/mod.rs

```diff
@@ -10,6 +10,27 @@
 
 #![macro_use]
 
+pub enum PseudoHeaderIPv6IPv4 {
+    Ipv6PseudoHeader,
+    Ipv4PseudoHeader,
+}
+
+/// Used to help compute checksum value of packet
+pub trait PseudoHeader {
+    /// Converts IPv4 or IPv6 specific fields to a checksum value
```

Comment:
Perhaps, with this being IP specific, it might be better in `pnet::packet::ip`, rather than `pnet::packet`.

In fact - since it's only implemented for IPv4 and IPv6, is it worth having a trait at all? You may as well just implement them separately, and special case them. That removes any question of returning an enum or using an associated type, and allows you to have IPv4/IPv6 specific parameters.